src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -49,16 +37,34 @@
 import games.strategy.util.CompositeMatchAnd;
 import games.strategy.util.Match;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
 /**
  * Pro non-combat move AI.
  */
 public class ProNonCombatMoveAI {
 
+  private final ProOddsCalculator calc;
   private GameData data;
```

Comment:
Bit of a copy paste smell here between ProCombatMove and ProNonCombatMove, they have the same internal data, I would guess the two have a lot more in common than just that.. Perhaps out of scope, but something to clean up..